<launch>
    <arg name="gui" default="true" />
    <arg name="SCENE" default="01" />

    <arg name="INPUT_PANORAMA_IMAGE" default="/dual_fisheye_to_panorama/output" />
    <arg name="INPUT_PANORAMA_INFO" default="/dual_fisheye_to_panorama/panorama_info" />
    <arg name="INPUT_CLASS" default="/edgetpu_panorama_object_detector/output/class" />
    <arg name="INPUT_RECTS" default="/edgetpu_panorama_object_detector/output/rects" />

    <param name="/use_sim_time" value="true" />
    <include file="$(find jsk_perception)/sample/include/play_rosbag_panorama_camera.xml">
        <arg name="SCENE" value="$(arg SCENE)" />
    </include>

    <!-- visualize output of object detector -->
    <node pkg="jsk_perception" type="draw_rects" name="draw_rects">
        <remap from="~input" to="$(arg INPUT_PANORAMA_IMAGE)" />
        <remap from="~input/rects" to="$(arg INPUT_RECTS)" />
        <remap from="~input/class" to="$(arg INPUT_CLASS)" />
    </node>

    <!-- rect_array_in_panorama_to_bounding_box_array -->
    <node pkg="jsk_perception" type="rect_array_in_panorama_to_bounding_box_array.py" name="rect_array_in_panorama_to_bounding_box_array">
        <remap from="~panorama_image" to="$(arg INPUT_PANORAMA_IMAGE)" />
        <remap from="~panorama_info" to="$(arg INPUT_PANORAMA_INFO)" />
        <remap from="~input_rects" to="$(arg INPUT_RECTS)" />
        <remap from="~input_class" to="$(arg INPUT_CLASS)" />
        <rosparam>
        frame_fixed: "insta360_link"
        dimensions_labels:
            person: [0.5, 0.5, 1.5]
    </rosparam>
    </node>

    <!-- motpy_bbox_tracker -->
    <node pkg="jsk_perception" type="motpy_bbox_tracker_node.py" name="motpy_bbox_tracker_node" output="screen">
        <remap from="~input" to="/rect_array_in_panorama_to_bounding_box_array/bbox_array" />
        <rosparam subst_value="true">
        dt: 0.2
        target_labels:
          - person
        label_names: $(find jsk_perception)/config/coco_labelnames.yaml
        approximate_sync: True
        slop: 0.1
    </rosparam>
    </node>

    <node pkg="rviz" type="rviz" name="$(anon rviz)" if="$(arg gui)" args="-d $(find jsk_perception)/sample/config/sample_motpy_bbox_tracker.rviz" />
</launch>
